script/apps/Aries/Creator/Game/Entity/RegionContainer.lua

```diff
@@ -78,23 +78,42 @@ function RegionContainer:SaveToFile(filename)
 	end
 	
 	local root = {name='entities', attr={file_version="0.1"} }
+	local sortEntities = {};
+	
 	for entity in pairs(self.entities) do 
 		if( entity:IsPersistent() and entity:IsRegional() and not entity:IsDead()) then
 			local node = {name='entity', attr={}};
-			entity:SaveToXMLNode(node);
-			if(node) then
-				root[#root+1] = node;
-			end
+			entity:SaveToXMLNode(node, true);
+			local x, y, z = entity:GetBlockPos();
+
+			table.insert(sortEntities, {pos = {x, y, z}, node = node});
```

Comment:
please precompute pos_value when inserting to sortEntities.  BTW,  test if x is nil.